covalent_ecs_plugin/ecs.py

```diff
@@ -129,7 +125,7 @@ def __init__(
         )
         self.vcpu = vcpu or get_config("executors.ecs.vcpu")
         self.memory = memory or get_config("executors.ecs.memory")
-        self._cwd = tempfile.mkdtemp()
+        self._ecs_task_family_name = ""
```

Comment:
Thanks for including this much-needed change re: `_cwd` @FyzHsn , while we are at it, can we move `L133` which creates the cache_dir to outside the constructor of the executor for self-hosted support? Either the top-most line in the `run` method works or if we create a getter method which also creates it if it doesn't exist. TIA!